### ISSUE TITLE
Bot crashes without warning if #lion_project_github does not exist

### ISSUE THREAD

**Issue by LambdaPaul** [cs-discord-at-ucf/lion#697]
Need some sort of null check for the channel during bot init.

**Comment by LambdaPaul** [comment]
Forgot to mention that this is not an issue locally but on docker/podman.

**Comment by LambdaPaul** [comment]
Correction, during prod mode.

**Comment by tanndlin** [comment]
This is intended, a design decision I made a while ago. Since Lion is run on a server of a static set of channels and only intended for the UCF CS server. 

The other alternative is to null check *every time* we need to look up a channel, which is unnecessary since each test server should be made off the template

**Comment by LambdaPaul** [comment]
I only made this issue because for the other channels it does not seem to care and shoots only a warning log instead of straight up crashing without saying why.

**Comment by tanndlin** [comment]
> I only made this issue because for the other channels it does not seem to care and shoots only a warning log instead of straight up crashing without saying why.

It's doing that because when you change to production mode, the bot try's to call GuildService.getChannel for the GitHub channel

If you use any plugin/handler that requires this, the same will happen for any other channel

I do agree there should be a warning, and an error thrown, but I do not want to have to null check.

**Comment by LambdaPaul** [comment]
Maybe not a null check, but some sort of logging would suffice in case there is an accidental deletion of a channel.

**Comment by c650** [comment]
yeah it would be nice to say "expected channel to exist" or something? hopefully we catch a toplevel exception for it. someone else can tal though

**Comment by tanndlin** [comment]
Would be some simple one liner [here](https://github.com/cs-discord-at-ucf/lion/blob/47276134cf3773300d852affc823b229a08b4aff/src/services/guild.service.ts#L52-L58)

Currently the bot just asserts the type instead of null checking
